src/banman.cpp

```diff
@@ -159,12 +135,43 @@ bool BanMan::Unban(const CSubNet& sub_net)
     return true;
 }
 
-void BanMan::GetBanned(banmap_t& banmap)
+void BanMan::UnbanAll(const CNetAddr& net_addr)
+{
+    {
+        LOCK(m_cs_banned);
+        std::vector<CSubNet> entries_to_delete;
+        for (const auto& it : m_banned) {
+            CSubNet sub_net = it.first;
+            if (sub_net.Match(net_addr)) entries_to_delete.push_back(sub_net);
+        }
+
+        if (!entries_to_delete.empty()) m_is_dirty = true;
+        for (const auto&it : entries_to_delete) {
+            m_banned.erase(it);
+        }
+    }
+
+    if (m_client_interface) m_client_interface->BannedListChanged();
+    DumpBanlist(); //store banlist to disk immediately
+}
+
+void BanMan::GetBanned(banmap_t& banmap, Optional<const CNetAddr&> filterForIp)
 {
     LOCK(m_cs_banned);
     // Sweep the banlist so expired bans are not returned
     SweepBanned();
     banmap = m_banned; //create a thread safe copy
+
+    if (filterForIp) {
```

Comment:
same, I wonder if this could just be a `std::remove_if`

Reply:
Done.